src/TabPane.js

```diff
@@ -170,6 +184,10 @@ let TabPane = React.createClass({
       labelledBy = getId(eventKey, TAB) || null;
     }
 
+    if (!visible && !Transition && this.getUnmountOnExit()) {
```

Comment:
Maybe move this up to before line 162? No reason to run through the logic above when it doesn't have any effect.